activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -501,10 +500,8 @@ def initialize_type_map(m) # :nodoc:
           m.alias_type 'lseg', 'varchar'
           m.alias_type 'box', 'varchar'
 
-          m.register_type 'timestamp' do |_, _, sql_type|
-            precision = extract_precision(sql_type)
-            OID::DateTime.new(precision: precision)
-          end
+          register_class_with_precision m, 'time',      OID::Time
+          register_class_with_precision m, 'timestamp', OID::DateTime
```

Comment:
Didn't I already do this? Am I going crazy? Also I thought I killed the time class.